src/Microsoft.AspNet.OData.Shared/Routing/Conventions/EntitySetRoutingConvention.cs

```diff
@@ -24,21 +23,6 @@ public partial class EntitySetRoutingConvention
         /// </returns>
         internal static string SelectActionImpl(ODataPath odataPath, IWebApiControllerContext controllerContext, IWebApiActionMap actionMap)
         {
-            if (odataPath == null)
```

Comment:
why remove these checks?

Reply:
Because it's am internal method, the checks are done in the public version of the method.